### PR TITLE
fix git repo issues

### LINKED ISSUE #59: Support using the splitter from repo to repo
### Current behavior

The splitter has issues currently functioning on one git repo to another. I don't know the details, but Jason explained it had to do with discoverability of the adoc content from the root of the dir (or something). So currently can only run the splitter locally. For the splitter's purpose, running from git repo to git repo is the primary if not the sole use case for it, so working that out is will be essential.

### Desired behavior

Be able to run the spliiter successfully from a git repo source to a git repo target, whether that be any of the following variations:
* GitHub > GitLab
* GitHub > GitHub
* GitHub repo branch > same repo, diff branch
* GitLab > GitHub
* GitLab > GitLab

The last two are very unlikely, but I figure if we can just get the splitter working on whatever repo and branch, with any credentials needed, then it shouldn't matter much. This is already the target behavior today based on the splitter options available, but just not quite working yet.

### PR DESCRIPTION
**Pull request by manaswinidas**
Remove extra title and context from assembly files and Fixes #59

### CODE REVIEW

File: src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java [hunk 1]
```diff
@@ -38,6 +38,9 @@
     @Option(names = {"-a", "--attributes"}, description = "Attributes to pass to asciidoctor")
     Map<String, Object> attributes;
 
+    @Option(names = {"-i", "--ignore"}, description = "Ignore file")
+    String ignoreFile;
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java:42]: I think a list of files is actually better.

File: src/main/java/com/redhat/documentation/asciidoc/extraction/Assembly.java [hunk 2]
```diff
@@ -46,10 +46,7 @@ public Assembly(Document doc, List<String> lines) {
         this.createAssembly = Boolean.parseBoolean(doc.getAttribute("assembly", "true").toString());
 
         // Adding the id of the module
-        this.source.append("[id=\"").append(this.idWithoutContext).append("_{context}\"]\n")
-                // Adding the section title
-                .append("= ").append(doc.getTitle()).append("\n")
-                .append(":context: ").append(this.idWithoutContext).append("\n\n");
+        this.source.append("[id=\"").append(this.idWithoutContext).append("_{context}\"]\n");
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/extraction/Assembly.java:49]: Why??

File: src/main/java/com/redhat/documentation/asciidoc/extraction/Extractor.java [hunk 3]
```diff
@@ -60,6 +60,11 @@ public void process() {
         final Path targetDirPath = this.task.getPushableLocation().getDirectoryPath().normalize();
 
         for (File file : new AsciiDocDirectoryWalker(sourceDirPath.toString())) {
+            // Skip ignore files
+            if (file.getParent() != null && file.getName().equals(task.getIgnoreFile())) {
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/extraction/Extractor.java:64]: The getParent check here is not necessary.

File: src/main/java/com/redhat/documentation/asciidoc/extraction/model/Task.java [hunk 4]
```diff
@@ -11,15 +11,17 @@
     private final Location location;
     private final PushableLocation pushableLocation;
     private final Map<String, Object> attributes;
+    private final String ignoreFile;
 
     public Task(Location location, PushableLocation pushableLocation) {
-        this(location, pushableLocation, Collections.emptyMap());
+        this(location, pushableLocation, Collections.emptyMap(), null);
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/extraction/model/Task.java:17]: Pass an empty string instead. But with a list an empty collection would be better.

File: src/main/java/com/redhat/documentation/asciidoc/extraction/Assembly.java (hunk 2 above)
Comment (manaswinidas) [reply src/main/java/com/redhat/documentation/asciidoc/extraction/Assembly.java:49]: the title and context lines were getting repeated

File: src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java [hunk 5]
```diff
@@ -38,6 +39,9 @@
     @Option(names = {"-a", "--attributes"}, description = "Attributes to pass to asciidoctor")
     Map<String, Object> attributes;
 
+    @Option(names = {"-i", "--ignore"}, description = "Ignore file")
+    List<String> ignoreFiles;
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/cli/ExtractionRunner.java:43]: You could make this a File or a Path. Might make it easier to use later.

File: src/main/java/com/redhat/documentation/asciidoc/extraction/Assembly.java (hunk 2 above)
Comment (LightGuard) [reply src/main/java/com/redhat/documentation/asciidoc/extraction/Assembly.java:49]: Got it

File: src/main/java/com/redhat/documentation/asciidoc/extraction/model/Task.java [hunk 6]
```diff
@@ -43,6 +46,10 @@ public PushableLocation getPushableLocation() {
         return Map.copyOf(attributes);
     }
 
+    public List<String> getIgnoreFiles() {
+        return ignoreFiles;
```
Comment (LightGuard) [inline src/main/java/com/redhat/documentation/asciidoc/extraction/model/Task.java:50]: This should be a copy of the array so it stays immutable

**Pull request merged**